src/lib.rs

```diff
@@ -622,6 +623,13 @@ impl<T: Counter> Histogram<T> {
             1
         } - 1;
         let sub_bucket_count = 2_usize.pow(sub_bucket_half_count_magnitude as u32 + 1);
+
+        if unit_magnitude + sub_bucket_half_count_magnitude + 1 > 63 {
+            // Cannot represent shifted sub bucket count in 64 bits.
+            // This will cause an infinite loop when calculating number of buckets
+            return Err("Cannot represent significant figures' worth of measurements beyond lowest value");
```

Comment:
I'm pretty sure that the original Java is vulnerable to this issue as well, but probably nobody uses minimum values that are very very large.